app/assets/javascripts/thyme/thyme_feedback.js

```diff
@@ -34,34 +34,35 @@ $(document).on('turbolinks:load', function() {
   seekBar.add();
 
   // heatmap
-  const heatmap = new Heatmap('heatmap', ['presentation', 'content', 'note']);
+  const heatmap = new Heatmap('heatmap');
 
   // below-area
-  const toggleMistakeAnnotations = new AnnotationCategoryToggle(
-    'toggle-mistake-annotations', 'mistake', null); // <- don't draw mistake annotations in the heatmap
-  const togglePresentationAnnotations = new AnnotationCategoryToggle(
-    'toggle-presentation-annotations', 'presentation', heatmap);
-  const toggleContentAnnotations = new AnnotationCategoryToggle(
-    'toggle-content-annotations', 'content', heatmap);
-  const toggleNoteAnnotations = new AnnotationCategoryToggle(
-    'toggle-note-annotations', 'note', heatmap);
-  toggleMistakeAnnotations.add();
-  togglePresentationAnnotations.add();
-  toggleContentAnnotations.add();
-  toggleNoteAnnotations.add();
-  const toggles = [toggleMistakeAnnotations, togglePresentationAnnotations,
-                   toggleContentAnnotations, toggleNoteAnnotations];
+  const allCategories = Category.all();
+  const annotationCategoryToggles = new Array(allCategories.length);
+  let category;
+
+  for (let i = 0; i < allCategories.length; i++) {
+    category = allCategories[i];
+    annotationCategoryToggles[i] = new AnnotationCategoryToggle(
+      'toggle-' + category.name + '-annotations', category, heatmap
+    );
+    if (category === Category.MISTAKE) {
+      // exclude mistake annotations from heatmap
+      annotationCategoryToggles[i].heatmap = null;
+    }
+    annotationCategoryToggles[i].add();
+  }
```

Comment:
I wanted to have the same indices in `allCategories` and `annotationCategoryToggles`. I guess, it will have the correct ordering with a for-of loop, but I'm a little bit careful.

Reply:
sorry, I didn't see you had two arrays here, so a for loop with an index is actually needed. Another alternative would be to iterate over `allCategories.entries()` as seen [here](https://stackoverflow.com/a/51683785/9655481), but a for loop with in index does the exact same thing, so it's fine.